src/plugins/shared/clusterOptions/selectClusterOptions.ts

```diff
@@ -0,0 +1,91 @@
+import { QuickPickItem, QuickPickItemKind, window } from "vscode";
+import { ReadyAzureSessionProvider } from "../../../auth/types";
+import { ClusterPreference } from "../types";
+import { Errorable } from "../../../commands/utils/errorable";
+import { RecentCluster } from "./state/recentCluster";
+import { selectExistingClusterOption } from "./options/selectExistingClusterOption";
+import { selectNewClusterOption } from "./options/selectNewClusterOption";
+import { selectRecentClusterOption } from "./options/selectRecentClusterOption";
+
+enum Options {
+    RecentCluster = "RecentCluster",
+    ExistingCluster = "ExistingCluster",
+    NewCluster = "NewCluster"
+}
+
+export type QuickPickClusterOptions = QuickPickItem & { type: Options }
+
+const getClusterOptions = () => {
+    const options = [
+        {
+            label: "", //seperator
+            type: Options.RecentCluster
+        },
+        {
+            label: "Recently used cluster",
+            type: Options.RecentCluster
+        },
+        {
+            label: "Existing cluster from your subscription",
+            type: Options.ExistingCluster
+        },
+        {
+            label: "Create new cluster from AKS VS Extension",
+            type: Options.NewCluster
+        }
+    ];
+
+    return options;
+}
+
+export async function selectClusterOptions(sessionProvider: ReadyAzureSessionProvider): Promise<Errorable<ClusterPreference | boolean>> {
+    const options = getClusterOptions();
+
+    const doesTempClusterExist = await RecentCluster.doesTempClusterExist();
```

Comment:
please include try/catch for proper error handling here as well.

Reply:
This has been updated with try/catch